### PR TITLE
Split reducers logic

### LINKED ISSUE #334: Split rootReducer into multiple reducers
As @HugoGresse [suggests](https://github.com/Instant-Visio/InstantVisio-WebApp/pull/324#discussion_r542366133)

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
Fix #334

### CODE REVIEW

File: src/components/App/userReducer.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { JWTToken } from '../../../types/JWT'
+import { SET_TOKEN } from '../../actions/types'
+
+const initialState = {
+    user: { token: '' },
+}
+
+export interface User {
+    token: JWTToken
```
Comment (HugoGresse) [inline src/components/App/userReducer.ts:9]: shoudll'nt the userId be also stored there, as well as the login state?

File: src/components/App/userSelector.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { createSelector } from 'reselect'
+import { User } from './userReducer'
+
+export const selectToken = createSelector(
+    (state: any) => state.user,
```
Comment (HugoGresse) [inline src/components/App/userSelector.ts:5]: not an any state

File: src/components/App/userReducer.ts (hunk 1 above)
Comment (ntchou) [reply src/components/App/userReducer.ts:10]: I work on it 
Check in my pr
https://github.com/Instant-Visio/InstantVisio-WebApp/pull/330/commits/f71a540879b7a7c0b53a566610488c6b25f90fdd

**Review by HugoGresse — APPROVED** [review]

File: src/components/App/userReducer.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply src/components/App/userReducer.ts:10]: Sorry, I forgot to assign myself [the issue](https://github.com/Instant-Visio/InstantVisio-WebApp/issues/334), and did not see the notification of your assignment. But wasn't your PR only meant to address `Redirect admin dashboard` ?

**Pull request merged**